monix-execution/shared/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -60,12 +62,34 @@ object Local {
   def clearContext(): Unit =
     localContext.set(defaultContext())
 
-  /** Execute a block of code without propagating any `Local.Context`
+  /** Execute a synchronous block of code without propagating any `Local.Context`
     * changes outside.
     */
   def isolate[R](f: => R): R =
     macro Macros.isolate
 
+  /** Execute an asynchronous block of code without propagating any `Local.Context`
+    * changes outside.
+    */
+  def isolate[R](f: => Future[R]): Future[R] = {
+    // TODO: rewrite as macro
+    val prev = Local.getContext()
+    val current = Local.getContext().mkIsolated
+    Local.setContext(current)
+
+    try {
+      f.transform(result => {
```

Comment:
Can't you use `Try[A] => Try[B]` overload here to have just one lambda?

Reply:
I think it's not available in `2.11`

Reply:
Oh, right

Reply:
I guess we can always create `Future.transform` in version specific modules to optimize it for `2.12` and `2.13` but not sure if it matters right now

Reply:
Actually, I think we have `transform` in `FutureUtils`, I'll try it